test/unit/dom/insertBefore.js

```diff
@@ -0,0 +1,195 @@
+import canPatchNativeAccessors from '../../../src/util/can-patch-native-accessors';
+import create from '../../lib/create';
+
+describe('dom: insertBefore', function () {
+  runTests('div');
+  runTests('slot');
+  runTests('host');
+  runTests('root');
+
+  function runTests(type) {
+    describe(`${type}: `, () => {
+      let host, root, slot, div, elem;
+
+      beforeEach(function () {
+        host = document.createElement('div');
+        root = host.attachShadow({ mode: 'open' });
+        slot = create('slot');
+
+        root.appendChild(slot);
+
+        div = document.createElement('div');
+
+        switch(type) {
+          case 'div':
+            elem = div;
+            break;
+          case 'slot':
+            elem = slot;
+            break;
+          case 'root':
+            root.innerHTML = '';
+            elem = root;
+            break;
+          default:
+            elem = host;
+        }
+      });
+
+      it(`should not insert anything if there is nothing to insert`, () => {
```

Comment:
the description doesn't match what the test does - maybe change test description to `should throw if nothing to insert`? would be good to also check the child count